src/sentry/notifications/helpers.py

```diff
@@ -406,6 +407,32 @@ def collect_groups_by_project(groups: Iterable[Group]) -> Mapping[int, set[Group
     return projects
 
 
+def get_user_subscriptions_for_groups_v2(
```

Comment:
@snigdhas Not a huge fan of having to pass in `notification_settings` here, could we just query the values internally?